src/js/components/TransformationSettings.react.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+const React = require('react');
+
+import { Accordion, Icon } from 'semantic-ui-react';
+import AdsTypes from '../data/AdsTypes';
+import SettingsActions from '../data/SettingsActions';
+
+import type { Props as BaseProps } from '../containers/AppContainer.react';
+
+type Props = BaseProps & {
+  accordionActive: boolean,
+  accordionIndex: number,
+  onAccordionTitleClick: (e: Event, itemProps: AccordionItemProps) => void
+};
+
+type AccordionItemProps = {
+  index: number
+};
+
+class TransformationSettings extends React.Component<Props> {
+  handleAdsRawHtmlChanged = (event: SyntheticEvent<HTMLInputElement>) => {
+    SettingsActions.editAdsRawHtml(event.currentTarget.value);
+  };
+
+  handleAdsTypeChanged = (event: SyntheticEvent<HTMLInputElement>) => {
+    SettingsActions.editAdsType(event.currentTarget.value);
+  };
+
+  handleAnalyticsRawHtmlChanged = (event: SyntheticEvent<HTMLInputElement>) => {
+    SettingsActions.editAnalyticsRawHtml(event.currentTarget.value);
+  };
+
+  handleAudienceNetworkPlacementIdChanged = (
+    event: SyntheticEvent<HTMLInputElement>
+  ) => {
+    SettingsActions.editAudienceNetworkPlacementId(event.currentTarget.value);
+  };
+
+  handleFbPixelIdChanged = (event: SyntheticEvent<HTMLInputElement>) => {
+    SettingsActions.editFbPixelId(event.currentTarget.value);
+  };
+
+  handleStyleNameChanged = (event: SyntheticEvent<HTMLInputElement>) => {
+    SettingsActions.editStyleName(event.currentTarget.value);
+  };
+
+  renderAudienceNetworkDiv() {
+    return this.props.settings.adsSettings.type ===
+      AdsTypes.AUDIENCE_NETWORK ? (
+        <div>
+          <label className="sub-label" htmlFor="audienceNetworkId">
+            <span>
+              {this.props.settings.adsSettings.audienceNetworkPlacementId
+                ? '✔'
+                : '•'}
+            </span>&nbsp; Audience Network ID
+          </label>
+          <div className="field">
+            <input
+              type="text"
+              name="audienceNetworkId"
+              placeholder="123456"
+              onChange={this.handleAudienceNetworkPlacementIdChanged}
+              value={this.props.settings.adsSettings.audienceNetworkPlacementId}
+            />
+          </div>
+        </div>
+      ) : null;
+  }
+
+  renderAdsRawHtmlDiv() {
+    return this.props.settings.adsSettings.type === AdsTypes.RAW_HTML ? (
+      <div>
+        <label className="sub-label" htmlFor="adsRawHtml">
+          <span>{this.props.settings.adsSettings.rawHtml ? '✔' : '•'}</span>&nbsp;
```

Comment:
I think we should have constants for these values ✔and • as we are using them in a lot of places. Or maybe a helper function where you pass the value and it renders one or the other based on if it's null or not.
As this is just a general improvement, feel free to open a new issue to address this.

Reply:
Sounds good! Thanks